cmd/oras/internal/display/status/interface.go

```diff
@@ -64,6 +64,14 @@ type CopyHandler interface {
 	OnMounted(ctx context.Context, desc ocispec.Descriptor) error
 }
 
+// ManifestPushHandler handles status output for manifest push command.
+type ManifestPushHandler interface {
+	OnManifestExists(desc ocispec.Descriptor) error
+	OnManifestUploading(desc ocispec.Descriptor) error
+	OnManifestUploaded(desc ocispec.Descriptor) error
```

Comment:
In the scope of `manifest push`, descriptor should be obtained before status output. Consider make it a property of `ManifestPushHandler` rather than pass it as a parameter for handler functions.